localstack/aws/handlers/service_plugin.py

```diff
@@ -37,32 +39,31 @@ def require_service(self, _: HandlerChain, context: RequestContext, response: Re
         if not context.service:
             return
 
+        service_name: str = context.service.service_name
+        if not self.service_manager.exists(service_name):
+            raise NotImplementedError
+
+        service_operation: Optional[ServiceOperation] = context.service_operation
         request_router = self.service_request_router
 
-        # verify that we have a route for this request
-        service_operation = context.service_operation
+        # Ensure the Service is loaded and set to ServiceState.RUNNING if not in an erroneous state.
+        service_plugin: Service = self.service_manager.require(service_name)
+
+        # Continue adding service skelethon and handlers to the router if these are missing.
         if service_operation in request_router.handlers:
             return
 
         with self.service_locks[context.service.service_name]:
             # try again to avoid race conditions
             if service_operation in request_router.handlers:
                 return
-
-            service_name = context.service.service_name
-            if not self.service_manager.exists(service_name):
-                raise NotImplementedError
-
-            service_plugin: Service = self.service_manager.require(service_name)
-
             if isinstance(service_plugin, Service):
                 if type(service_plugin.listener) == AwsApiListener:
                     request_router.add_skeleton(service_plugin.listener.skeleton)
                 else:
                     request_router.add_handler(service_operation, LegacyPluginHandler())
             else:
                 LOG.warning(
-                    "found plugin for %s, but cannot attach service plugin of type %s",
-                    service_name,
-                    type(service_plugin),
+                    f"found plugin for '{service_name}', "
+                    f"but cannot attach service plugin of type '{type(service_plugin)}'",
                 )
```

Comment:
generally we do not use f-strings for logging statements to defer string creation to the actual log statement (which may not actually happen at all - saving a bit of cpu time). in this case it will probably not have a huge impact, but for the sake of consistency we should keep the normal call format.